iree/tools/init_xla_dialects.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This files defines a helper to trigger the registration of dialects to
+// the system.
+//
+// Based on MLIR's InitAllDialects but without dialects we don't care about.
+
+#ifndef IREE_TOOLS_INIT_XLA_DIALECTS_H_
+#define IREE_TOOLS_INIT_XLA_DIALECTS_H_
+
+#include "mlir/IR/Dialect.h"
+#include "tensorflow/compiler/mlir/xla/ir/chlo_ops.h"
+#include "tensorflow/compiler/mlir/xla/ir/hlo_ops.h"
+#include "tensorflow/compiler/mlir/xla/ir/lhlo_ops.h"
+
+namespace mlir {
+
+// This function should be called before creating any MLIRContext if one expect
+// all the possible dialects to be made available to the context automatically.
+inline void registerXLADialects() {
+  static bool init_once = []() {
+    registerDialect<mlir::xla_hlo::XlaHloDialect>();
+    registerDialect<mlir::xla_chlo::XlaHloClientDialect>();
+    registerDialect<mlir::xla_lhlo::XlaLhloDialect>();
```

Comment:
Do we use LHLO? I thought we didn't. If you want to leave it in for now and then try deleting it later, that's also fine with me :-)

Reply:
This is just a copy&paste from TF's [dialect_registration.cc](https://github.com/tensorflow/tensorflow/blob/dfc190e78833f4d619657b908f9aeec8c4f09b0a/tensorflow/compiler/mlir/xla/ir/dialect_registration.cc#L24). Honestly, didn't checked if it is really used, but the `lhlo_ops` are at least included in the CMake tensorflow target. So, if we drop this we should do some more refactoring, maybe in a followup.

Reply:
:+1: